pandas/core/arrays/boolean.py

```diff
@@ -282,14 +284,25 @@ def _from_sequence(
 
     @classmethod
     def _from_sequence_of_strings(
-        cls, strings: List[str], *, dtype: Optional[Dtype] = None, copy: bool = False
+        cls,
+        strings: List[str],
+        *,
+        dtype: Optional[Dtype] = None,
+        copy: bool = False,
+        true_values: Optional[List[str]] = None,
+        false_values: Optional[List[str]] = None,
     ) -> "BooleanArray":
+        if true_values is not None:
+            cls.TRUE_VALUES.update(true_values)
```

Comment:
I think we need to make a copy here of TRUE_VALUES and not update it in place? (one call should not influence a next call)

For example something like

```
if true_values is not None:
    true_values = TRUE_VALUES.union(true_values)
else:
    true_values = TRUE_VALUES
```

Reply:
Changed. Was not sure what to prefer here.